File: bank_api_integration/bank_api_integration/report/bank_statement_ledger/bank_statement_ledger.js
// Copyright (c) 2016, Aerele and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Bank Statement Ledger"] = {
	"filters": [
		{
			"fieldname":"account_no",
			"label": __("Account No"),
			"fieldtype": "Data",
			"read_only": 1
		},
		{
			"fieldname":"date",
			"label": __("Date"),
			"fieldtype": "Data",
			"read_only": 1
		},
		{
			"fieldname":"balance",
			"label": __("Balance"),
			"fieldtype": "Data",
			"read_only": 1
		},
		{
			"fieldname":"currency",
			"label": __("Currency"),
			"fieldtype": "Data",
			"read_only": 1
		}
	],
	onload: function(frm)
	{
	frappe.call({
	method: "bank_api_integration.bank_api_integration.doctype.bank_api_integration_settings.bank_api_integration_settings.fetch_balance",
	freeze: true,
	callback: function(r) {
		if(r.message) {
		frappe.query_report.set_filter_value('account_no',r.message['account_no']);
		frappe.query_report.set_filter_value('date',r.message['date']);
		frappe.query_report.set_filter_value('balance',r.message['balance']);
		frappe.query_report.set_filter_value('currency',r.message['currency']);
		}
	}
		});
	}
};

File: bank_api_integration/bank_api_integration/report/bank_statement_ledger/bank_statement_ledger.py
# Copyright (c) 2013, Aerele and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns, data = [], []
	columns = [{'label': 'Date', 'fieldname': 'date', 'fieldtype': 'Data', 'options': None, 'width': 180},
	{'label': 'Transaction ID', 'fieldname': 'transaction_id', 'fieldtype': 'Data', 'options': None, 'width': 180},
	{'label': 'Credit', 'fieldname': 'credit', 'fieldtype': 'Currency', 'options': 'Currency', 'width': 180},
	{'label': 'Debit', 'fieldname': 'debit', 'fieldtype': 'Currency', 'options': 'Currency', 'width': 180},
	{'label': 'Description', 'fieldname': 'description', 'fieldtype': 'Data', 'options': None, 'width': 180}]
	data = frappe.get_list("Bank Transaction", fields=["date", "transaction_id", "credit", "debit", "description"])
	return columns, data
